common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -456,3 +509,64 @@ func appendError(aggregatedErr error, err error) error {
 	}
 	return fmt.Errorf("%v, %w", aggregatedErr, err)
 }
+
+func (s *localStoreCertProvider) refreshCerts() {
+
+	for {
+		select {
+		case <-s.stop:
+			break
+		case <-s.ticker.C:
+		}
+
+		newCerts, err := s.loadCerts()
+		if err != nil {
+			s.logger.Error("failed to load certificates", tag.Error(err))
+			continue
+		}
+
+		if s.certs.isEqual(newCerts) {
+			continue
+		}
+
+		s.RWMutex.Lock()
+		s.certs = newCerts
+		s.RWMutex.Unlock()
+	}
+}
+
+func (c *certCache) isEqual(other *certCache) bool {
```

Comment:
it is possible that either 'c' or 'other' can be nil right?

Reply:
Added checks.